File: cp_rust/src/bin/abc265-a-apple.rs
// Created by Ayush Biswas at 2026/07/07 20:30
// https://atcoder.jp/contests/adt_easy_20260617_1/tasks/abc265_a
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        [x, y, n]: [usize; 3]
    ) -> usize {
        if 3*x <= y {
            n * x
        } else {
            (n/3)*y + (n%3)*x
        }
    }
}
// @code end

File: cp_nim/src/bin/abc265-a-apple.nim
# Created by Ayush Biswas at 2026/07/18 18:39
# https://atcoder.jp/contests/adt_easy_20260617_1/tasks/abc265_a
import ../lib/cpio

# @code begin

var r = newReader()

let (x, y, n) = r[int, int, int]

if 3 * x <= y:
  echo n * x
else:
  let total = (n div 3) * y + (n mod 3) * x
  echo total

# @code end
